contracts/bprotocol/avatar/AbsCToken.sol

```diff
@@ -153,43 +162,47 @@ contract AbsCToken is Cushion {
         address payable userOrDelegatee
     ) external onlyBToken postPoolOp(true) returns (uint256) {
         uint256 result = cToken.borrow(borrowAmount);
+        if(result != 0) return result;
+
+        _score().updateDebtScore(address(this), address(cToken), toInt256(borrowAmount));
+
+        // send funds at last
         if(_isCEther(cToken)) {
-            // FIXME OZ `Address.sendValue`
-            userOrDelegatee.transfer(borrowAmount);
+            bool success = userOrDelegatee.send(borrowAmount);
+            success; //shh: avoid DoS attack
         } else {
             IERC20 underlying = cToken.underlying();
             underlying.safeTransfer(userOrDelegatee, borrowAmount);
         }
-        _score().updateDebtScore(address(this), address(cToken), toInt256(borrowAmount));
         return result;
     }
 
     // ERC20
     // ======
     function transfer(ICToken cToken, address dst, uint256 amount) public onlyBToken postPoolOp(true) returns (bool) {
-        bool result = cToken.transfer(dst, amount);
+        address dstAvatar = registry.getAvatar(dst);
+        bool result = cToken.transfer(dstAvatar, amount);
         uint256 underlyingRedeemAmount = _toUnderlying(cToken, amount);
         _score().updateCollScore(address(this), address(cToken), -toInt256(underlyingRedeemAmount));
```

Comment:
technically if result if false, we should not do it. maybe just revert if false?